platform-sdk/platform-apps/tests/MigrationTestingTool/src/main/java/com/swirlds/demo/migration/MigrationTestingToolState.java

```diff
@@ -278,6 +282,17 @@ public void init(
         }
     }
 
+    @Override
+    public void preHandle(
+            @NonNull Event event,
+            @NonNull Consumer<ScopedSystemTransaction<StateSignatureTransaction>> stateSignatureTransaction) {
+        event.forEachTransaction(transaction -> {
+            if (!transaction.isSystem() && isSystemTransaction(transaction.getApplicationTransaction())) {
```

Comment:
You can extract the isSystem() check as a separate code before the if statement and add a comment on top of it - about why we check whether it's the deprecated system transaction format